sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1518,6 +1518,14 @@ object SQLConf {
       .intConf
       .createOptional
 
+  val HIVE_FUNCTION_RESOURCE_URI_LENGTH_THRESHOLD =
+    buildConf("spark.sql.hive.functionResourceUriLengthThreshold")
+      .internal()
+      .doc("The maximum length allowed for a function resource uri in Hive metastore.")
+      .version("4.0.0")
+      .intConf
+      .createWithDefault(4000)
```

Comment:
Just a question. Could you give us a reference for this limit?

Reply:
And, why this is a configurable?

Reply:
This limit is from the default Hive Metastore SQL schema
```RESOURCE_URI` VARCHAR(4000) CHARACTER SET latin1 COLLATE latin1_bin,```
So the 4000 characters is the out-of-the-box limit for each function resource entry.

Actually it's better to make this hardcoded, since it's extermely rare that people might want to change this configuration.

Reply:
+1 to define a constant variable and use it~

Reply:
BTW, `latin1` length sounds different from a Spark's unicode length URI. Is there a chance of any issues in the future?

Reply:
Most ASCII-based characters should be fine. I don’t expect utilization of non-ASCII characters (e.g i18n languages) except in parameter and/or function comments, so this should be unlikely.